### PR TITLE
Updated the replace method for the sanitize method.

### PR DESCRIPTION
**Pull request by Ryanc1256**
Just updated the random replace methods with the actual sanitize method.

### CODE REVIEW

**Comment by devinivy** [comment]
Just a reference of the sanitize method for those code-reviewing:
```node
/**
 * Sanitize a key removing any spaces or reserved charaters
 *
 * @param {String} str
 */

utils.sanitize = function sanitize(str) {
  return typeof str === 'string' ? str.replace(/\s+/g, '_') : str;
};
```

**Comment by devinivy** [comment]
The case-insensitive flag is definitely not necessary.  The only difference I can think of is that multiple white space characters will now be condensed into a single whitespace character.

```node
// Old code
('white   space').replace(/\s/gi, '_') == 'white___space'

// New code
('white   space').replace(/\s+/g, '_') == 'white_space'

**Comment by Ryanc1256** [comment]
That is true. You think i should make it condensing the single space character.

**Comment by devinivy** [comment]
I think the safest option is to _not_ condense the whitespace character, since it could allow for new key collisions (`key 1` and `key   1`) beyond those that are already possible (`key 1` and `key_1`).  But I'm not sure if it's a practical concern– you'd probably know better than I would.  Should we try to get another person to give their opinion.

**Comment by Ryanc1256** [comment]
I totally agree, didn't pass my mind haha, okay ill change it and push it up again, just with out the condensing of the characters.

File: lib/utils.js [hunk 1]
```diff
@@ -57,6 +57,16 @@ utils.sanitize = function sanitize(str) {
 };
 
 /**
+ * Sanitize a key by removing spaces or reserved charaters
+ *
+ * @param {String} str
+ */
+
+utils.removeSpaces = function removeSpaces(str) {
```
Comment (eddieajau) [inline lib/utils.js]: Only comment is that the function name is not strictly correct. Perhaps a better name is `replaceSpaces`, and then allowing an optional argument for the replacement. For example:

```js
/**
 * Replaces whitespace with a character or string.
 *
 * @param {string} str - source string
 * @param {string} [replacement] - character or string to replace spaces with. Default: "_"
 * @return {string}
 */
utils.removeSpaces = function removeSpaces(str, replacement) {
  return typeof str === 'string' ? str.replace(/\s/g, replacement || '_') : str;
};
```

**Comment by eddieajau** [comment]
@Ryanc1256, can you add a unit test for this? (if you need help with that, let me know)

**Comment by Ryanc1256** [comment]
@eddieajau Definitely!

**Comment by Ryanc1256** [comment]
@eddieajau there we go, is that better?

**Comment by Ryanc1256** [comment]
@eddieajau is it good to merge this? or do you want me to change more?

**Pull request merged**